example/viam_example_app/lib/screens/stream.dart

```diff
@@ -18,14 +22,59 @@ class StreamScreen extends StatefulWidget {
 }
 
 class _StreamScreenState extends State<StreamScreen> {
+  // Stream
   bool _ready = false;
   late RTCVideoRenderer _renderer;
   late async.StreamSubscription<MediaStream> _streamSub;
 
+  // Single frame
+  ByteData? imageBytes;
+  bool _imgLoaded = false;
+
+  void _getImage() {
+    setState(() {
+      _imgLoaded = false;
+    });
+    final imageFut = widget.camera.getImage();
+    imageFut.then((value) {
+      final convertFut = convertImageToFlutterUi(value.image ?? img.Image.empty());
+      convertFut.then((value) {
+        final pngFut = value.toByteData(format: ui.ImageByteFormat.png);
+        pngFut.then((value) => setState(() {
+              imageBytes = value;
+              _imgLoaded = true;
+            }));
+      });
+    });
+  }
```

Comment:
[q] is there a reason you favored `then` over `async / await` in this method?

Reply:
Mostly because I don't know how to use async/await when called from UI code... :)

Reply:
cool cool, ok no worries. We can refactor this later, to align patterns, not a big deal for now.

There are a couple of ways to handle it, the primary way could be with a [FutureBuilder<T>](https://api.flutter.dev/flutter/widgets/FutureBuilder-class.html) - it does require some reworking of the UI though. there are also some unofficial packages/widgets that we could use as well.